internal/services/keyvault/key_vault_key_resource.go

```diff
@@ -436,12 +435,12 @@ func resourceKeyVaultKeyUpdate(d *pluginsdk.ResourceData, meta interface{}) erro
 		return err
 	}
 
-	if v, ok := d.GetOk("rotation_policy"); ok {
-		if respPolicy, err := client.UpdateKeyRotationPolicy(ctx, id.KeyVaultBaseUrl, id.Name, expandKeyVaultKeyRotationPolicy(v)); err != nil {
+	if d.HasChange("rotation_policy"); ok {
+		if respPolicy, err := client.UpdateKeyRotationPolicy(ctx, id.KeyVaultBaseUrl, id.Name, expandKeyVaultKeyRotationPolicy(d.Get("rotation_policy"))); err != nil {
```

Comment:
we should cast this before passing it in, since this should always be _something_ (even if that's an empty slice)

```suggestion
		if respPolicy, err := client.UpdateKeyRotationPolicy(ctx, id.KeyVaultBaseUrl, id.Name, expandKeyVaultKeyRotationPolicy(d.Get("rotation_policy").([]interface{})); err != nil {
```

(it might well be there's a bug in the Read function where an empty slice isn't set into the state for `rotation_policy`, which would be a bug to fix fwiw)